### ISSUE TITLE
push zuluplay to the repositories | replace tcplay

### ISSUE THREAD

**Issue by DJCrashdummy** [mhogomchungu/zuluplay#1]
are you planing to distribute `zuluplay` via the repositories for debian/ubuntu/fedora?

i guess it will also make sense to replace `tcplay` with a transitional package to `zuluplay`, since it is still at version 1.1 (even in debian sid and ubuntu artful) and the project itself seems dead for a long time now.

i know there is also a [veracrypt-fork of tcplay](https://github.com/veracrypt/tc-play), but it seems also abandoned... do you/somebody have more knowledge regarding this topic?

**Comment by mhogomchungu** [comment]
tcplay only supports TrueCrypt volumes.
veracrypt fork of tcplay added support for VeraCrypt volumes.Lets call this fork veraplay for simplicity.
zuluplay started as a fork of veraplay and added support for PIM value in VeraCrypt volumes. I created the fork because maintainer of veraplay did not want to make changes that i needed in zuluCrypt.

It also extended tcplay API to essentially push zuluCrypt code in zuluplay.

zuluplay was originally a bundled package of zuluCrypt but i created a separate project for it because some distributions have policies against bundled projects. It currently supports all features of tcplay and hence it can be used as its replacement but a bit of work is needed to add VeraCrypt support its the CLI interface. I mainly use its library and i have put no effort so far in extending its CLI interface. 

In tcplay and veraplay, an unlocked volume creates an entry in "/dev/disk/by-id/" that is in a format of
```dm-uuid-CRYPT-TCPLAY-foo```

in zuluCrypt,i needed a way to distinguish between an unlocked VeraCrypt volume and an unlocked TrueCrypt volume and i broke the format in zuluplay by creating ```dm-uuid-CRYPT-TCRYPT-foo``` for a TrueCrypt volume and ```dm-uuid-CRYPT-VCRYPT-foo``` volume for a VeraCrypt volume. As far as the outside world is concerned. This is the only difference that sets zuluplay apart from the others when it comes to managing TrueCrypt volumes since currently the CLI API only supports TrueCrypt volumes..

**Comment by DJCrashdummy** [comment]
wow! thank you very much for the interesting view behind the curtains.

hmm... i don't see why veracrypt rejected your changes. since truecrypt- and veracrypt-volumes are not completely compatible, i think it is not that bad to distinguish between them.

so... as far as i see, it would make sense to "replace" `tcplay` with `zuluplay` in future.

**Comment by mhogomchungu** [comment]
Within a week or two,i will complete the CLI interface and make a formal release of zuluplay and distributions should start using that released version.

**Comment by DJCrashdummy** [comment]
this is awesome to hear! :-)

**Comment by mhogomchungu** [comment]
The git version now has a complete CLI interface.

An example command to unlock a TrueCrypt volume is below:
```
zuluplay -m foo -d /dev/sdc1
```
An example command to unlock a VeraCrypt volume is below:
```
zuluplay -m foo -d /dev/sdc1 --veracrypt-mode
```

An example command to unlock a VeraCrypt volume that use a PIM value is below:
```
zuluplay -m foo -d /dev/sdc1 --veracrypt-mode --pim 3
```
The --veracrypt-mode option is not necessary since it is assumed when a pim option is given.

**Comment by mhogomchungu** [comment]
Just released version 1.1 of zuluplay that has full CLI support for unlocking TrueCrypt volumes,VeraCrypt volumes and VeraCrypt volumes that use a PIM value.

It will be up to distribution to pick up zuluplay or replace tcplay with it, I am not a packager and i have no say in what packagers do.

**Comment by DJCrashdummy** [comment]
FYI @mhogomchungu there is some progress...

Laszlo Boszormenyi is the [maintainer of the tcplay-package](https://packages.debian.org/sid/tcplay) and Marcio de Souza Oliveira is the [maintainer of the zulucrypt-/zulumount-packages](https://packages.debian.org/sid/zulucrypt-cli) in debian.
since i'm not really familiar with fedora i have no clue about the progress there.

> On 2017-06-17 17:28, Marcio de Souza Oliveira wrote:
> > Hi Laszlo, Walter
> > 
> > I will create the ITP bug for zuluplay because the source the zulucrypt
> > not build the zuluplay binary,
> > the zulucrypt uses the internal version of "tcplay", that is "zuluplay",
> > at least part of the zuluplay.
> > 
> > [1] https://github.com/mhogomchungu/zuluplay/releases
> > 
> > I believe that tcplay can be replace by zuluplay without problems, so I
> > will package zuluplay.
> > 
> > Regards,
> > 
> > Marcio
> > 
> > On 15-06-2017 07:45, László Böszörményi (GCS) wrote:
> >> Hi Walter, Marcio,
> >>
> >> On Wed, Jun 14, 2017 at 4:31 PM, Walter Dietrich wrote:
> >>> -) tcplay 2.0 was released 3 years ago (but now it seems dead)
> >>
> >>   Yes, this is known.
> >>
> >>> -) veracryt forked tcplay (https://github.com/veracrypt/tc-play) to add
> >>> veracrypt-support (but it also seems dead)
> >>
> >>   Also known.
> >>
> >>> -) last but not least...
> >>> the developer of zulucrypt/zulumount - which are already in the
> >>> repositories - created zuluplay which is a active fork of tcplay with
> >>> veracrypt-support and more...
> >>
> >>   Heard about it, but never tried. I switched to file based disc
> >> cryptography a while. Then I mostly stopped using it.
> >>
> >>> this may be interesting and clarify some things:
> >>> https://github.com/mhogomchungu/zuluplay/issues/1
> >>>
> >>> is it possible and makes it sense to add or replace tcplay with zuluplay
> >>> like mentioned in the github-issue above?
> >>
> >>   Can be, thanks for the heads-up. If zuluplay 1.1 or later will be
> >> part of Sid, I am going to ask its maintainer to replace tcplay.
> >>
> >>> hopefully i could help a little bit and not just bothered you!
> >>
> >>   Thanks, it was nice from you.
> >>
> >> Kind regards,
> >> Laszlo/GCS
> >
